### PR TITLE
Consistent output file names in all backends

### PR DESCRIPTION
**Pull request by Julow**
The latex and man backends were prefixing names differently than the HTML backend. This can create naming conflicts.

This removes code that was caused problems while working on https://github.com/ocaml/odoc/pull/1190

### CODE REVIEW

File: src/document/url.mli [hunk 1]
```diff
@@ -27,6 +27,9 @@ module Path : sig
 
   val string_of_kind : kind -> string
 
+  val pp_kind_prefix_for_output : Format.formatter -> kind -> unit
+  (** Print the ["kind-"] prefix used in output files. *)
```
Comment (panglesd) [inline src/document/url.mli:31]: This seems to be for output file in html flat mode, but if I understand correctly used as labels in latex mode (and not as filenames).

Maybe:

```suggestion
  val pp_disambiguating_prefix : Format.formatter -> kind -> unit
  (** Print the ["kind-"] prefix used to disambiguate urls in "flat modes": e.g. latex labels and output files in [--flat] HTML output *)
```

**Review by panglesd — COMMENTED** [review]
I left a small comments, otherwise the implementation looks good to me.

Note that this changes labels in latex output. Do we have any guarantee that those won't change, just like we try not modifying URLs in HTML output? Can there be any handwritten  latex referencing odoc labels?

File: src/document/url.mli (hunk 1 above)
Comment (Julow) [reply src/document/url.mli]: And the man output too. Thanks!

File: CHANGES.md [hunk 2]
```diff
@@ -42,6 +42,8 @@
 - Allow `][` in code blocks (@Julow, #1149)
   This was interpreted as "code blocks with result", which now mandate a delimiter:
   `{delim@lang[ code ]delim[ result ]}`
+- Output file paths in the man and latex backends changed to avoid name clashes
```
Comment (panglesd) [inline CHANGES.md:45]:
```suggestion
- Output file paths and labels in the man and latex backends changed to avoid name clashes
```
Just so that if someone is using the latex backend and handwriting labels, they can find the cause of their new problem in the changelog.

**Review by panglesd — APPROVED** [review]
LGTM (apart from the CI)!

File: src/html/generator.ml [hunk 3]
```diff
@@ -476,7 +476,7 @@ end
 
 module Page = struct
   let on_sub = function
-    | `Page _ -> None
+    | `Pge _ -> None
```
Comment (panglesd) [inline src/html/generator.ml:479]: Is this change intended?

Comment (Julow) [reply src/html/generator.ml]: Fixed, thanks
